static/app/views/performance/transactionSummary/transactionEvents/utils.tsx

```diff
@@ -0,0 +1,60 @@
+import {LocationDescriptor, Query} from 'history';
+
+import {Organization} from 'app/types';
+import {TableDataRow} from 'app/utils/discover/discoverQuery';
+import {generateEventSlug} from 'app/utils/discover/urls';
+import {getTraceDetailsUrl} from 'app/views/performance/traceDetails/utils';
+
+import {getTransactionDetailsUrl} from '../../utils';
+
+export function eventsRouteWithQuery({
+  orgSlug,
+  transaction,
+  projectID,
+  query,
+}: {
+  orgSlug: string;
+  transaction: string;
+  query: Query;
+  projectID?: string | string[];
+}) {
+  const pathname = `/organizations/${orgSlug}/performance/summary/events/`;
+  return {
+    pathname,
+    query: {
+      transaction,
+      project: projectID,
+      environment: query.environment,
+      statsPeriod: query.statsPeriod,
+      start: query.start,
+      end: query.end,
+      query: query.query,
+    },
+  };
+}
+
+export function generateTraceLink(dateSelection) {
+  return (
+    organization: Organization,
+    tableRow: TableDataRow,
+    _query: Query
+  ): LocationDescriptor => {
+    const traceId = `${tableRow.trace}`;
+    if (!traceId) {
+      return {};
+    }
+
+    return getTraceDetailsUrl(organization, traceId, dateSelection, {});
+  };
+}
+
+export function generateTransactionLink(transactionName: string) {
+  return (
+    organization: Organization,
+    tableRow: TableDataRow,
+    query: Query
+  ): LocationDescriptor => {
+    const eventSlug = generateEventSlug(tableRow);
+    return getTransactionDetailsUrl(organization, eventSlug, transactionName, query);
+  };
+}
```

Comment:
Any reason why these functions are now duplicated here and [here](https://github.com/getsentry/sentry/blob/26344147c1fa3f8236f904d90a042556f17f8e1b/static/app/views/performance/transactionSummary/content.tsx#L397) rather than sharing an import?

Reply:
nice catch 😄  I've removed the duplicate and imported from `utils`. Also moved the code up a level to `transactionSummary/utils.tsx` since it made more sense.